booster/library/Booster/Syntax/Json/Internalise.hs

```diff
@@ -619,10 +666,11 @@ isTermM pat = case pat of
    the arguments as a list. The top-level sorts of the 'And' nodes are
    ignored, no checks are performed.
 -}
-explodeAnd :: Syntax.KorePattern -> [Syntax.KorePattern]
-explodeAnd Syntax.KJAnd{patterns} =
-    concatMap explodeAnd patterns
-explodeAnd other = [other]
+explodeAnd :: Syntax.KorePattern -> NE.NonEmpty Syntax.KorePattern
+explodeAnd Syntax.KJAnd{patterns} = case patterns of
+    [] -> error "malformed KJAnd"
```

Comment:
I think the benefit of having this function returning `NonEmpty` and exploding on an invalid And is not big, as we only use the non-empty in `isTop`/`isBottom`. I think we can just as well use the normal list there and avoid the call to error?